vivarium/core/engine.py

```diff
@@ -337,23 +465,66 @@ def __init__(
         log.info(pf(self.topology))
 
 
+    def _add_step_path(
+            self,
+            step: Step,
+            path: HierarchyPath,
+            # None if deriver, empty list if no dependencies.
+            relative_dependencies: Optional[Sequence[HierarchyPath]],
+            depend_on_derivers: bool = False,
```

Comment:
What is this `depend_on_derivers` used for? It looks like deriver-like steps are added in `_add_process_path` by not having `relative_dependencies`.

Reply:
This describes whether the first step should be added with a dependency on any existing derivers. Now that I'm writing this, I realize that we need to to the topological sorting first so that we know which step(s) come first. Supporting derivers this way might be difficult...

Reply:
Let's keep the backwards compatibility of derivers as simple as possible. I think just pull all steps/derivers out of `processes` and put them in `self.step_paths`. If they are not included in `flow`, then assume deriver-like behavior. We can take the last of these deriver-steps, and make the first flow-steps depend on it -- or make them depend on all deriver-steps. Alternatively, don't even put dependence between deriver-steps and flow-steps, and we'll deal with issues if they come up.

Reply:
done. I'm not holding the derivers in a separate list in `StepGraph` so that it's easy to return those as the first layers